### PR TITLE
Fix dependency detection for zopfli support

### PR DESCRIPTION
**Pull request by simonihmig**
The previous attempt of fixing zopfli support did actually not work in real life, just in tests! 😝
See the inline comments for explanation. Fixed it, including tests, and tested manually.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -131,12 +136,7 @@ module.exports = {
       },
 
       _hasPackage(pkgName) {
-        try {
-          this.project.require.resolve(pkgName);
```
Comment (simonihmig) [inline index.js]: Ember CLI seems to just offer `require`, without any of its members that you would normally expect. So `require.resolve()` was always throwing, leading to the function returning false.

File: tests/index-test.js [hunk 2]
```diff
@@ -157,7 +157,17 @@ describe('compress plugin', function() {
           'assets/ignore.js',
         ],
         ui: mockUi,
-        project: { name: function() { return 'test-project'; }, require },
```
Comment (simonihmig) [inline tests/index-test.js]: The mock of `project.require` here was actually providing the "full" `require` API, so this was passing in tests.

File: tests/index-test.js [hunk 3]
```diff
@@ -157,7 +157,17 @@ describe('compress plugin', function() {
...
+        project: {
+          name() {
+            return 'test-project';
+          },
+          require() {
+            return require(...arguments);
```
Comment (simonihmig) [inline tests/index-test.js]: Only provide `require` without any of its members in tests.

**Comment by simonihmig** [comment]
@cibernox thanks for the quick merge! Could you cut a patch release with this?

**Comment by cibernox** [comment]
@simonihmig it's released already!

**Comment by simonihmig** [comment]
@cibernox you're the best! 🙏

**Pull request merged**